### PR TITLE
Getrf tuning 2

### PR DESCRIPTION
**Pull request by jzuniga-amd**
This PR adds further optimizations to GETRF and adds internal implementations of GEMM and TRSM to improve the performance of the factorizations of small and medium-size matrices.

### CODE REVIEW

**Comment by tfalders** [comment]
getrf and getrf_npvt seem to be experiencing test failures on CI with this PR. These will need to be addressed before I can review the code.

File: library/src/lapack/roclapack_getrf.hpp [hunk 1]
```diff
@@ -67,345 +110,155 @@ inline rocblas_int getrf_get_innerBlkSize(rocblas_int m, rocblas_int n, const bo
...
+    //    {
+    //    }
+    //    else
+    //    {
+    //    }
+    //}
+    //else
+    //{
+    //    if(pivot)
+    //    {
```
Comment (tfalders) [inline library/src/lapack/roclapack_getrf.hpp:129]: Is there a reason to keep these comments around?

File: library/src/lapack/roclapack_getrf.hpp [hunk 2]
```diff
@@ -544,89 +393,61 @@ rocblas_status rocsolver_getrf_template(rocblas_handle handle,
...
-        // <===== (LOOP FACTORIZING INNER BLOCKS)
+        // factorize inner block panel
+        getrf_panelLU<BATCHED, STRIDED, T>(handle, m - j, jb, n, A, shiftA + j, lda, strideA, ipiv,
+                                           shiftP + j, strideP, info, batch_count, pivot, scalars,
+                                           work1, work2, work3, work4, optim_mem, pivotval,
+                                           pivotidx, j, iipiv, m);
 
         // update trailing matrix
-        if(j + jb < n)
+        nextpiv = j + jb; //posicion for the matrix update
```
Comment (tfalders) [inline library/src/lapack/roclapack_getrf.hpp:419]: Spelling: position

**Review by tfalders — APPROVED** [review]
LGTM! Just some minor comments.

File: library/src/lapack/roclapack_getrf.hpp (hunk 1 above)
Comment (jzuniga-amd) [reply library/src/lapack/roclapack_getrf.hpp:129]: Yes, I need to do fine-tuning for the block sizes in the batched and npvt cases, but for now, as this is needed asap, I will merge like this and then just fill in the blanks in a future PR

**Pull request merged**